common/src/main/java/org/opensearch/ml/common/transport/model/MLUpdateModelInput.java

```diff
@@ -0,0 +1,177 @@
+    
+    public static final String MODEL_ID_FIELD = "model_id"; // mandatory
+    public static final String DESCRIPTION_FIELD = "description"; // optional
+    public static final String MODEL_NAME_FIELD = "name"; // optional
+    public static final String MODEL_GROUP_ID_FIELD = "model_group_id"; // optional
+    public static final String MODEL_CONFIG_FIELD = "model_config"; // optional
+    public static final String CONNECTOR_ID_FIELD = "connector_id"; // optional
+
+    @Getter
+    private String modelId;
+    private String description;
+    private String name;
+    private String modelGroupId;
+    private MLModelConfig modelConfig;
+    private String connectorId;
+
+    @Builder(toBuilder = true)
+    public MLUpdateModelInput(String modelId, String description, String name, String modelGroupId, MLModelConfig modelConfig, String connectorId) {
+        this.modelId = modelId;
+        this.description = description;
+        this.name = name;
+        this.modelGroupId = modelGroupId;
+        this.modelConfig = modelConfig;
+        this.connectorId = connectorId;
+    }
+
+    public MLUpdateModelInput(StreamInput in) throws IOException {
+        this.modelId = in.readString();
+        this.description = in.readOptionalString();
+        this.name = in.readOptionalString();
+        this.modelGroupId = in.readOptionalString();
+        if (in.readBoolean()) {
+            modelConfig = new TextEmbeddingModelConfig(in);
+        }
+        this.connectorId = in.readOptionalString();
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.field(MODEL_ID_FIELD, modelId);
+        if (description != null) {
+            builder.field(DESCRIPTION_FIELD, description);
+        }
+        if (name != null) {
+            builder.field(MODEL_NAME_FIELD, name);
+        }
+        if (modelGroupId != null) {
+            builder.field(MODEL_GROUP_ID_FIELD, modelGroupId);
+        }
+        if (modelConfig != null) {
+            builder.field(MODEL_CONFIG_FIELD, modelConfig);
+        }
+        if (connectorId != null) {
+            builder.field(CONNECTOR_ID_FIELD, connectorId);
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeString(modelId);
+        out.writeOptionalString(description);
+        out.writeOptionalString(name);
+        out.writeOptionalString(modelGroupId);
+        if (modelConfig != null) {
+            out.writeBoolean(true);
+            modelConfig.writeTo(out);
+        } else {
+            out.writeBoolean(false);
+        }
+        out.writeOptionalString(connectorId);
+    }
+
+    public static MLUpdateModelInput parse(XContentParser parser, String modelId) throws IOException {
+        String description = null;
```

Comment:
maybe have this call the parse method without `modelId` and then set `modelId` in the result?

Reply:
What do you mean by "set `modelId` in the result?"